### PR TITLE
Fix wait-for-snapshot-create action

### PR DESCRIPTION
**Pull request by probakowski**
Even if snapshot reports success through GetSnapshotStatus API it might still not finished at that point (it means only that all shards are in the store but not that finalization went ok in cluster state).
This change adds additional check to `wait-for-snapshot-create` action that checks if target snapshot is in list of currently running snapshot with GetSnapshot `_current` API

### CODE REVIEW

**Review by dliappis — COMMENTED** [review]
Left a few initial comments. Note there are also a few test failures.

File: esrally/driver/runner.py [hunk 1]
```diff
@@ -1846,6 +1846,16 @@ async def __call__(self, es, params):
...
+            response = await es.snapshot.get(repository=repository, snapshot="_current", verbose=False)
+            running = False
+            for s in response["snapshots"]:
+                if s["snapshot"] == snapshot:
+                    running = True
+                    break
+            if running:
+                await asyncio.sleep(wait_period)
+                continue
+
```
[the remark this answers is not in the document]
Comment (probakowski) [reply esrally/driver/runner.py:1853]: I'd say it's more readable, I've updated it (it should be `if snapshot in` though as we want to loop if the snapshot is running currently - is in the list)

**Comment by probakowski** [comment]
@dliappis I've changed the code according to your suggestion and fixed tests

**Review by dliappis — APPROVED** [review]
LGTM. I did a bunch of additional testing using this PR using restore challenges of the logging use case and everything seems to work fine.

**Comment by probakowski** [comment]
Thank you @dliappis !

**Pull request merged**